song-server/src/main/java/bio/overture/song/server/service/ValidationService.java

```diff
@@ -109,6 +118,43 @@ public Optional<String> validate(@NonNull JsonNode payload) {
     return Optional.ofNullable(errors);
   }
 
+  private void externalValidations(
+      List<ExternalValidation> externalValidations, @NonNull JsonNode payload) {
+
+    for (ExternalValidation externalValidation : externalValidations) {
+      String jsonPath = externalValidation.getJsonPath();
+      String externalUrl = externalValidation.getUrl();
+
+      // Extract the value from the uploaded JSON using the jsonPath
+      String value = JsonPath.read(payload, "$." + jsonPath);
+
+      if(Objects.isNull(value) || value.isEmpty())
+        return;
+
+      // call the external url using restTemplate
+    }
+  }
+
+  public boolean invokeExternalUrl(String studyId, String url, String value) {
```

Comment:
Don't see where are we calling this method.

Reply:
We are not calling this method as of now, because the external API is not up and running due to which all the validations will fail.